packages/react-accordion/src/components/Accordion/useAccordion.ts

```diff
@@ -1,51 +1,38 @@
 import * as React from 'react';
-import { makeMergeProps, resolveShorthandProps } from '@fluentui/react-utilities';
-import { AccordionProps, AccordionShorthandPropsCompat, AccordionState } from './Accordion.types';
+import { resolveShorthand } from '@fluentui/react-utilities';
+import { AccordionProps, AccordionSlots, AccordionState } from './Accordion.types';
 import { useCreateAccordionContextValue } from './useAccordionContext';
 
 /**
  * Const listing which props are shorthand props.
  */
-export const accordionShorthandPropsCompat: AccordionShorthandPropsCompat[] = ['expandIcon', 'button', 'icon'];
-
-const mergeProps = makeMergeProps<AccordionState>({ deepMerge: accordionShorthandPropsCompat });
+export const accordionShorthandProps: Array<keyof AccordionSlots> = [];
 
 /**
  * Returns the props and state required to render the component
  * @param props - Accordion properties
  * @param ref - reference to root HTMLElement of Accordion
  * @param defaultProps - default values for the properties of Accordion
  */
-export const useAccordion = (
-  props: AccordionProps,
-  ref: React.Ref<HTMLElement>,
-  defaultProps?: AccordionProps,
-): AccordionState => {
-  const state = mergeProps(
-    {
-      ref,
-      collapsible: false,
-      multiple: false,
-      navigable: false,
-      context: {
-        navigable: false,
-        openItems: [],
-        requestToggle() {
-          /* noop */
-        },
-      },
-      descendants: [],
-      setDescendants() {
-        /* noop */
-      },
+export const useAccordion = (props: AccordionProps, ref: React.Ref<HTMLElement>): AccordionState => {
+  const initialState = {
+    ref,
+    collapsible: false,
+    multiple: false,
+    navigable: false,
+    ...props,
+    components: {
+      root: 'div',
     },
-    defaultProps && resolveShorthandProps(defaultProps, accordionShorthandPropsCompat),
-    resolveShorthandProps(props, accordionShorthandPropsCompat),
-  );
-
-  const [context, descendants, setDescendants] = useCreateAccordionContextValue(state);
-  state.context = context;
-  state.descendants = descendants;
-  state.setDescendants = setDescendants;
-  return state;
+    button: props.button ? resolveShorthand(props.button) : undefined,
+    expandIcon: props.expandIcon ? resolveShorthand(props.expandIcon) : undefined,
+    icon: props.icon ? resolveShorthand(props.icon) : undefined,
+  } as const;
+  const [context, descendants, setDescendants] = useCreateAccordionContextValue(initialState);
```

Comment:
IMO it looks that this hook creates a strange set of dependencies without obvious need 😥 I tried to refactor it locally to avoid `initialState`:

```ts
export const useAccordion = (props: AccordionProps, ref: React.Ref<HTMLElement>): AccordionState => {
  const [descendants, setDescendants] = useDescendantsInit<AccordionDescendant>();

  const index = React.useMemo(() => normalizeIndex(props.index), [props.index]);
  const [openItems, setOpenItems] = useControllableValue<number[], HTMLElement>(index!, () =>
    initializeUncontrolledOpenItems(props),
  );

  const requestToggle = useEventCallback((ev: React.MouseEvent<HTMLElement>, i: number) => {
    if (descendants[i]?.disabled === true) {
      return;
    }

    props.onToggle?.(ev, i);
    setOpenItems(previousOpenItems =>
      updateOpenItems(i, previousOpenItems!, {
        collapsible: props.collapsible,
        multiple: props.multiple,
      }),
    );
  });

  return {
    collapsible: false,
    multiple: false,
    navigable: false,
    ...props,

    components: {
      root: 'div',
    },
    ref,

    button: resolveShorthand(props.button),
    expandIcon: resolveShorthand(props.expandIcon),
    icon: resolveShorthand(props.icon),

    descendants,
    openItems,
    setDescendants,
    requestToggle,
  };
};
```

In this case we will get a single object. WDYT?

Reply:
IMO we should avoid `state.context` as a pattern in general as our `state` is mutable: for example, I can modify `state.openItems` but it will break the component as `state.context.openItems` are left unmodified:

```tsx
const state = { openedItems: [] }

state.openedItems = [1]
```

Reply:
The problem with that is that I'll have to move the context creation to the render function, and since the render function is not a hook this will required me to do some eslint disabling which is not ideal...